java/src/main/java/com/powsybl/python/dynamic/DynamicModelMapper.java

```diff
@@ -0,0 +1,136 @@
+     *
+     * @param staticId also determines the dynamic id of the element
+     */
+    public void addAlphaBetaLoad(String staticId, String parametersIds) {
+        dynamicModelList.add(() -> new LoadAlphaBeta(staticId, staticId, parametersIds));
+    }
+
+    /**
+     * maps static element to a dynamic one transformer
+     *
+     * @param staticId also determines the dynamic id of the element
+     */
+    public void addOneTransformerLoad(String staticId, String parametersIds) {
+        dynamicModelList.add(() -> new LoadOneTransformer(staticId, staticId, parametersIds));
+    }
+
+    public void addOmegaRef(String generatorId) {
+        dynamicModelList.add(() -> new OmegaRef(generatorId));
+    }
+
+    public void addGeneratorSynchronousThreeWindings(String staticId, String parametersIds) {
+        dynamicModelList.add(() -> new GeneratorSynchronousThreeWindings(staticId, staticId, parametersIds));
+    }
+
+    public void addGeneratorSynchronousThreeWindingsProportionalRegulations(String staticId, String parametersIds) {
+        dynamicModelList.add(
+            () -> new GeneratorSynchronousThreeWindingsProportionalRegulations(staticId, staticId, parametersIds));
+    }
+
+    public void addGeneratorSynchronousFourWindings(String staticId, String parametersIds) {
+        dynamicModelList.add(() -> new GeneratorSynchronousFourWindings(staticId, staticId, parametersIds));
+    }
+
+    public void addGeneratorSynchronousFourWindingsProportionalRegulations(String staticId, String parametersIds) {
+        dynamicModelList.add(
+            () -> new GeneratorSynchronousFourWindingsProportionalRegulations(staticId, staticId, parametersIds));
+    }
+
+    public void addCurrentLimitAutomaton(String staticId, String parametersIds, Branch.Side side) {
+        dynamicModelList.add(() -> new CurrentLimitAutomaton(staticId, staticId, parametersIds, side));
+    }
+
+    /**
+     * Will add all the dynamic mapping contained in the dataframe.
+     *
+     * Handles default values for missing arguments like CURRENT_LIMIT_AUTOMATON.
+     * Ignores parameter_set_id for OMEGA_REF
+     *
+     * @param mappingDf dataframe that should respect
+     *                  {@link DynamicMappingsDataframeHandle#dynamicMappingsDataFrameMapper}
+     *                  format
+     */
+    public void addAllMappings(UpdatingDataframe mappingDf) {
+        StringSeries staticIdSeries = mappingDf.getStrings("static_id");
+        StringSeries parameterSetIdSeries = mappingDf.getStrings("parameter_set_id");
+        IntSeries dynamicModelSeries = mappingDf.getInts("mapping_type");
+        for (int row = 0; row < mappingDf.getRowCount(); ++row) {
+            DynamicMappingType type = DynamicMappingType.fromCValue(dynamicModelSeries.get(row));
+            String staticId = staticIdSeries.get(row);
+            String parameterSetId = parameterSetIdSeries.get(row);
+            switch (type) {
```

Comment:
Instead of this switch/case, we should imitate the facilities already in place for the network, with the `NetworkElementAdder` interface.

This should allow to have different columns for the current limit automaton, in particular.

Reply:
I have rewritten the whole part about dynamic mapping to work with adders like in networks.